### PR TITLE
Document GHC-12003

### LINKED ISSUE #231: TcRnTypeEqualityOutOfScope -> 12003

### LINKED ISSUE #323: TcRnTypeEqualityOutOfScope -> 12003

### PR DESCRIPTION
**Pull request by david-christiansen**
Fixes haskell/error-message-index#323.

### CODE REVIEW

File: message-index/messages/GHC-12003/index.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+title: Type equality not in scope
+summary: The type equality operator has not been imported into the current module
```
Comment (PPKFS) [inline message-index/messages/GHC-12003/index.md:3]:
```suggestion
summary: The type equality operator has not been imported into the current module.
```

File: message-index/messages/GHC-12003/index.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+severity: warning
+flag: "-Wcompat"
+introduced: 9.6.1
+---
+
+In prior versions of GHC, the type equality operator `~` was built-in syntax. In more recent versions, it is an ordinary operator that is part of the `Prelude` and of `Data.Type.Equality`. Restricting imports from `Prelude` can result in `~` not being imported. For now, GHC has a compatibility warning to help migrate old code. In a future version of GHC, this warning will be an error.
```
Comment (PPKFS) [inline message-index/messages/GHC-12003/index.md:9]: Could be useful to specify *which* versions

Comment (PPKFS) [reply message-index/messages/GHC-12003/index.md:9]: Have a safe trip!

Comment (david-christiansen) [reply message-index/messages/GHC-12003/index.md]: Thanks!

Updated.

**Review by PPKFS — APPROVED** [review]

**Pull request merged**